File: dbflow-core/src/main/java/com/raizlabs/android/dbflow/annotation/ModelViewQuery.java
package com.raizlabs.android.dbflow.annotation;

import com.raizlabs.android.dbflow.sql.Query;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Description: Represents a field that is a {@link Query}. This is only meant to be used as a query
 * reference in {@link ModelView}. This is so the annotation processor knows how to access the query of
 * the view.
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.SOURCE)
public @interface ModelViewQuery {
}

File: dbflow-core/src/main/java/com/raizlabs/android/dbflow/annotation/ModelViewQuery.kt
package com.raizlabs.android.dbflow.annotation

import com.raizlabs.android.dbflow.sql.Query

/**
 * Description: Represents a field that is a [Query]. This is only meant to be used as a query
 * reference in [ModelView]. This is so the annotation processor knows how to access the query of
 * the view.
 */
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.SOURCE)
annotation class ModelViewQuery
